studio/data/projects/project-readonly-status-query.ts

```diff
@@ -0,0 +1,55 @@
+import { useQuery, useQueryClient, UseQueryOptions } from '@tanstack/react-query'
+import { get } from 'lib/common/fetch'
+import { API_ADMIN_URL } from 'lib/constants'
+import { useCallback } from 'react'
+import { projectKeys } from './keys'
+
+export type ProjectReadOnlyStatusVariables = { projectRef?: string }
+export type ProjectReadOnlyStatus = {
+  enabled: boolean
+  override_enabled: boolean
+  override_active_until: string
+}
+
+export async function getProjectReadOnlyStatus(
+  { projectRef }: ProjectReadOnlyStatusVariables,
+  signal?: AbortSignal
+) {
+  if (!projectRef) throw new Error('Project ref is required')
+  const data = await get(`${API_ADMIN_URL}/projects/${projectRef}/readonly`, { signal })
```

Comment:
https://github.com/supabase/supabase/pull/15261/files#r1237982603

Due to the amount of requests we add, changing status to 'request changes' 😢